packages/apps/human-app/server/src/integrations/exchange-oracle/exchange-oracle.gateway.ts

```diff
@@ -2,128 +2,162 @@ import { Injectable } from '@nestjs/common';
 import { AxiosRequestConfig } from 'axios';
 import { lastValueFrom } from 'rxjs';
 import {
-  UserStatisticsDetails,
+  UserStatisticsCommand,
   UserStatisticsResponse,
 } from '../../modules/statistics/model/user-statistics.model';
 import { HttpService } from '@nestjs/axios';
 import {
-  OracleStatisticsDetails,
+  OracleStatisticsCommand,
   OracleStatisticsResponse,
 } from '../../modules/statistics/model/oracle-statistics.model';
 import {
+  JobAssignmentCommand,
   JobAssignmentData,
-  JobAssignmentDetails,
   JobAssignmentParams,
   JobAssignmentResponse,
   JobsFetchParams,
+  JobsFetchParamsCommand,
   JobsFetchParamsData,
-  JobsFetchParamsDetails,
   JobsFetchResponse,
+  ResignJobCommand,
+  ResignJobData,
 } from '../../modules/job-assignment/model/job-assignment.model';
 import {
   JobsDiscoveryParams,
+  JobsDiscoveryParamsCommand,
   JobsDiscoveryParamsData,
-  JobsDiscoveryParamsDetails,
   JobsDiscoveryResponse,
 } from '../../modules/jobs-discovery/model/jobs-discovery.model';
 import { Mapper } from '@automapper/core';
 import { InjectMapper } from '@automapper/nestjs';
-import { instanceToPlain } from 'class-transformer';
 import { HttpMethod } from '../../common/enums/http-method';
+import { toCleanObjParams } from '../../common/utils/gateway-common.utils';
+import { KvStoreGateway } from '../kv-store/kv-store.gateway';
+import { EscrowUtilsGateway } from '../escrow/escrow-utils-gateway.service';
 
 @Injectable()
 export class ExchangeOracleGateway {
```

Comment:
Why ExchangeOracleGateway and ReputationOracleGateway don't have the same structure? For example use getEndpointOptions

Reply:
Url for exchange oracle is received dynamically and in different ways for different endpoints (see postNewJobAssignment vs resignAssignedJob), as opposed to ones set for h-captcha/reputation oracle gateways - so we decided against alignment with other gateways